### DISCUSSION TITLE
Fail to build fluentui-blazor

### DISCUSSION THREAD

**Discussion by Woodman9** [microsoft/fluentui-blazor#1396] [General]
Here is what I'm seeing when I simply clone and build. How do I, or someone, fix this?

MSBuild version 17.8.3+195e7f5a3 for .NET
  Determining projects to restore...
  Restored C:\Users\woodm\source\repos\fluentui-blazor\src\Extensions\DesignToken.Generator\Microsoft.FluentUI.AspNetCore.Components.Generators.csproj (in 694 ms).
  9 of 10 projects are up-to-date for restore.
  
  > build
  > node ./esbuild.config.mjs

CSC : warning EnableGenerateDocumentationFile: Set MSBuild property 'GenerateDocumentationFile' to 'true' in project file to enable IDE0005 (Remove unnecessary usings/imports) on build (https://github.com/dotnet/roslyn
/issues/41640) [C:\Users\woodm\source\repos\fluentui-blazor\examples\Demo\DocGenerator\FluentUI.Demo.Generators.csproj]
CSC : warning EnableGenerateDocumentationFile: Set MSBuild property 'GenerateDocumentationFile' to 'true' in project file to enable IDE0005 (Remove unnecessary usings/imports) on build (https://github.com/dotnet/roslyn 
/issues/41640) [C:\Users\woodm\source\repos\fluentui-blazor\src\Extensions\DesignToken.Generator\Microsoft.FluentUI.AspNetCore.Components.Generators.csproj]
  file:///C:/Users/woodm/source/repos/fluentui-blazor/src/Core.Assets/esbuild.config.mjs:2
  import pkg from './package.json' assert { type: 'json' }
                                   ^^^^^^

  FluentUI.Demo.Generators -> C:\Users\woodm\source\repos\fluentui-blazor\examples\Demo\DocGenerator\bin\Debug\netstandard2.0\FluentUI.Demo.Generators.dll
  Microsoft.FluentUI.AspNetCore.Components.Generators -> C:\Users\woodm\source\repos\fluentui-blazor\src\Extensions\DesignToken.Generator\bin\Debug\netstandard2.0\Microsoft.FluentUI.AspNetCore.Components.Generators.dll 
  SyntaxError: Unexpected identifier
      at ESMLoader.moduleStrategy (node:internal/modules/esm/translators:139:18)
      at ESMLoader.moduleProvider (node:internal/modules/esm/loader:236:14)
      at async link (node:internal/modules/esm/module_job:67:21)
C:\Users\woodm\.nuget\packages\microsoft.visualstudio.javascript.sdk\0.5.401524-alpha\Sdk\Sdk.targets(172,5): error MSB3073: The command "npm run build" exited with code 1. [C:\Users\woodm\source\repos\fluentui-blazor\ 
src\Core.Assets\Microsoft.FluentUI.AspNetCore.Components.Assets.esproj]
  Microsoft.FluentUI.AspNetCore.Templates -> C:\Users\woodm\source\repos\fluentui-blazor\src\Templates\bin\Debug\net8.0\Microsoft.FluentUI.AspNetCore.Templates.dll

Build FAILED.

**Comment by dvoituron** [comment]
To work on this project and the `dev` branch, we are using:
- [Visual Studio Preview](https://visualstudio.microsoft.com/vs/preview/)
- [.NET 8 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
- NPM installed by [NodeJS](https://nodejs.org/en/download/)

Could you open your Terminal and run this command in the root folder: `dotnet restore`.
Then open Visual Studio to compile the solution.

**Reply by Woodman9** [reply]
I will thank you. I am running .NET 8 with VS Preview. I'm not sure if there is a particular version of NodeJS required. I'm running 16.13.1

**Reply by Woodman9** [reply]
It reports "All projects are up-to-date for restore."

**Reply by dvoituron** [reply]
I'm using NodeJS version v20.10.0.
Could you delete folders `src\Core.Assets\bin`, `obj` and `dist` and rebuild the **Core.Assets** project?

**Reply by Dovrol** [reply]
I recently built this project with just dotnet sdk 8.0, had similar error which was caused by old nodejs version. You probably just need to update node and everything should work fine

**Reply by Woodman9** [reply]
Cool. will give it a go. Thanks!
